File: csharp/bob/Bob.cs
using System;
using System.Text.RegularExpressions;
public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();


        if (statement == ""){
            return ("Fine. Be that way!");
        } 

        else if (Regex.IsMatch(statement,"[A-Z]") 
                && !Regex.IsMatch(statement, "[a-z]") 
                && statement.Substring(statement.Length - 1) != "?"){
                return ("Whoa, chill out!");
        }
        else if (Regex.IsMatch(statement,"[A-Z]") 
            && !Regex.IsMatch(statement, "[a-z]")
            && statement.Substring(statement.Length - 1) == "?"){
            return ("Calm down, I know what I'm doing!");
        }
        else if (Regex.IsMatch(statement, "[0-9]") 
                && statement.Substring(statement.Length - 1) == "?" 

                || Regex.IsMatch(statement, "[a-z]") 
                && statement.Substring(statement.Length - 1) == "?"
                
                || !Regex.IsMatch(statement, "[a-z]") 
                && statement.Substring(statement.Length - 1) == "?"
                && !Regex.IsMatch(statement, "[0-9]")) {
                return ("Sure.");
        }
        else if (statement == "Bob"){
            return ("Fine. Be that way!");
            
        }
        else {
            return ("Whatever.");
        }
    }
}

File: cpp/bob/bob.cpp
#include "bob.h"

using namespace std;

namespace bob {
    string hey(string str) {
        bool question, upperCase, lowerCase;
        question = upperCase = lowerCase = false;
        bool empty = true;
        
        for (char& c : str) {
            if (islower(c)) lowerCase = true;
            if (isupper(c)) upperCase = true;

            if (!isspace(c)) {
                question = false;
                empty = false;
            }

            if (c == '?') question = true;
        }
        
        bool shouting = upperCase && !lowerCase;
        if (shouting && question) return "Calm down, I know what I'm doing!";
        if (shouting) return "Whoa, chill out!";
        if (question) return "Sure.";
        if (empty) return "Fine. Be that way!";

        return "Whatever.";
    }
}  // namespace bob
